### DISCUSSION TITLE
Paste equivalent of Evergreen super menu functionality required

### DISCUSSION THREAD

**Discussion by mikaelaSmithTwilio** [twilio-labs/paste#3287] [Help: Figma] [answered]
In Evergreen we have a frequently used menu that supports dropdown behavior when clicked, but also allows search and collapsible categories within that menu. This is a menu style that is used frequently throughout the Segment product and I have not been able to identify a suitable Paste equivalent.

I have a loom video I can share, however this is for an unreleased product and I don't know that I can share it in this public forum. If someone can slack me directly I'd be happy to send you the screen recording with a walkthrough of the use case we need to be able to support.

Thanks!

**Comment by loreina** [comment]
Hi @mikaelaSmithTwilio,

Can you share a loom video in #help-design-system in Slack? Thanks!

**Comment by SiTaggart** [comment, marked as answer]
Hi @mikaelaSmithTwilio 

This component has come up a couple of times now. At the end of the day, it isn't easily made accessible, with very limited assistive technology support. The [ARIA spec](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/) does accommodate a dialog popup for a combobox, but it doesn't really work very well and results in a not great or clear experience for the user.

As a base principal, Paste leads with inclusive by default, and as such we don't recommend overly complex and complicated component constructs. We recommend re-visiting the interaction design and breaking it down into a composition of more base components that Paste provides out of the box.

We do provide a lot of base building blocks that can be used to create something that will achieve the same end result in terms of an interaction, but with a more simplistic collection of components.

Ultimately, we prioritise each users experience based on their ability and technology choice when creating our components, and as such we don't think this component makes a good candidate for the system and is not something we will likely visit in the future.

If the product team chooses to build something similar for their own app sub-system they are likely able to create something close using all our unstyled primitive components, and styling primitives. We won't recommend it, though, and won't guarantee the application's accessibility conformance.
